src/vs/workbench/api/browser/mainThreadTerminalService.ts

```diff
@@ -137,26 +139,25 @@ export class MainThreadTerminalService implements MainThreadTerminalServiceShape
 			extHostTerminalId,
 			isFeatureTerminal: launchConfig.isFeatureTerminal,
 			isExtensionOwnedTerminal: launchConfig.isExtensionOwnedTerminal,
-			useShellEnvironment: launchConfig.useShellEnvironment
+			useShellEnvironment: launchConfig.useShellEnvironment,
 		};
 		this._extHostTerminals.set(extHostTerminalId, new Promise(async r => {
-			let terminal: ITerminalInstance | undefined;
-			if (launchConfig.isSplitTerminal) {
-				const activeInstance = this._terminalService.getInstanceHost(launchConfig.target).activeInstance;
-				if (activeInstance) {
-					terminal = withNullAsUndefined(await this._terminalService.createTerminal({ instanceToSplit: activeInstance, config: shellLaunchConfig }));
-				}
-			}
-			if (!terminal) {
-				terminal = await this._terminalService.createTerminal({
-					config: shellLaunchConfig,
-					target: launchConfig.target
-				});
-			}
+			const terminal = await this._terminalService.createTerminal({
+				config: shellLaunchConfig,
+				location: await this._resolveMainThreadLocation(launchConfig.location)
```

Comment:
Let's always make `resolve*` the first thing we call in the cross process `$...` functions